src/v3/src/components/InputPassword/InputPassword.tsx

```diff
@@ -35,8 +39,20 @@ const InputPassword: UISchemaElementComponent<{
   } = uischema.options;
   const error = messages?.value?.[0] && getMessage(messages.value[0]);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  // For server side errors, need to reset the touched value
+  useEffect(() => {
+    setIsTouched(false);
+  }, [error]);
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setIsTouched(true);
     onChangeHandler(e.currentTarget.value);
+    onValidateHandler(setFieldError, e.currentTarget.value);
```

Comment:
what scenario this validation handles? iirc, the validation only happens in the confirm password field

Reply:
is it possible to avoid internal state / side effects from this component, and wrap those behavior in another hoc for rendering purpose, so this one can be used to compose other complex ones.

Reply:
State is needed, but it can be extracted to a HOC, however, will still need to reference those state props in each component so it knows when to show/hide the error message, so I will extract to a HOC.

Reply:
uischema controls that, you can manage `messages` field in schema programatically to control how message should be rendered.

Reply:
Unfortunately, it doesn't control it for live (on change/blur) errors, each component has to control it because it knows when the user touched the field and its current value.

Reply:
How about a stateless component as the base to handle or ui rendering logic, then a HOC for InputPassword (or give it another name), so it can be reused in PasswordWithConfirmation